test/README.md

```diff
@@ -0,0 +1,67 @@
+Candid test data
+================
+
+This directory contains Candid test data, to be used to check Candid
+implementations for compliance.
+
+This test suite covers
+
+ * parsing Candid service descriptions
+ * given a Candid type, decoding binary Candid data, asserting success or failure
+ * given a Candid type, decoding textual Candid data, asserting success or failure
+
+Because this test data necessarily has to be host-language agnostic, it is not
+possible to
+
+ * assert the decoded value, as it would be host-language specific
+ * cover encoding, as it would require specifying a host-language value,
+   and also because Candid encoding is not canonical
+
+Implementers are advised to extend this test suite with randomzied
+round-tripping tests (host value → candid → host value).
+
+Format
+======
+
+The test suite contains these kind of files:
+
+ * `<name>.good.did`:
+   A Candid service description, with valid candid syntax.
+   Implementations should be able to parse them.
+
+ * `<name>.bad.did`:
+   A Candid service description, with invalid candid syntax.
+   Implementations should reject parsing them.
+
+ * `<name>.test.did`:
+   A set of Candid tests, written in a grammar that extens the Candid type and value grammar:
+
+   The root production is
+   ```
+   <enctest>  ::= <def>;* <test>;*
+   ```
+   to allow any number of type definitions followed by encoding tests with grammar
+   ```
+   <test> = <argseq> <input> <outcome>
+   <argseq>  ::= ( <argtype>,* )
+   <input> =  <text> | blob <text>
+   <outcome> = true | false
```

Comment:
```suggestion
   <outcome> = fail | ok <text> | ok possible_blob <text>
```
This is more complete, and users are free to interpret `ok <text>` as `true` without further parsing.

Reply:
I didn't want to add additional keywords to the grammar, this might cause problems when trying to use the same lexer/parser for `.did` and `.test.did` files. We could use `variant { foo = … }`, but that’s such a mouthful (I wish we had `#foo … ` for variant values.)

What should `<text>` be? The round-tripped candid textual representation? Is that unique enough?

Reply:
It shouldn't be difficult to implement contextual keywords, parsed as identifiers.

Reply:
In that case ok